mark.c

```diff
@@ -876,9 +876,8 @@ GC_INNER mse * GC_mark_from(mse *mark_stack_top, mse *mark_stack,
             GC_ASSERT((word)limit >= (word)current_p);
             if (cheri_address_get(limit) < cheri_base_get(limit)) goto next_object;
   
-            has_rwx = cheri_perms_get(limit) & (CHERI_PERM_LOAD
-                                                | CHERI_PERM_STORE
-                                                | CHERI_PERM_EXECUTE);
+            has_rwx = cheri_perms_get(limit);
```

Comment:
Can you double check which format we decided to keep for this pattern? The commit for the associated discussion [1] is now (rightfully) deleted.

[1] https://github.com/capablevms/bdwgc/pull/19#discussion_r999502774

Reply:
The correct format has already been reverted to  in [PR#19](https://github.com/capablevms/bdwgc/pull/19). This is the change from before this PR was split.

Reply:
Is this a github quirk then? Because I read this as "this is how the code will change if this PR is merged", rather than "this is what the code was at the moment of raising this PR". If it's the latter, then I guess this diff is superfluous, and we can ignore it.

Reply:
Not really. I think if I merge this as is, it will be changed as you suspect. 
What I suggest is either, 
* that when finally force pushing, I will merge with master and correct it when resolving a merge conflict. --OR--
*  I can force push right now with this change taken out. 

@ltratt any other idea?  I do agree with @0152la that this sticking around is a little confusing.

Reply:
Wouldn't a commit on this PR which just changes the code here to be same as master make this diff go away? Then the commit would disappear upon squashing, and no force pushing at this stage required.

Reply:
Yes it would. I did not want to confuse the thread even further, but if this is OK. I can do it no problem!

Reply:
I think that's better personally than having to force push or fix future merge conflicts.

Reply:
Sure! Done in [b770233](https://github.com/capablevms/bdwgc/pull/18/commits/b7702331b8e102d9213d24c6e607822497a9fb6a)